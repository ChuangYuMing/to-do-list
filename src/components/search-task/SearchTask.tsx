import React, { Dispatch, SetStateAction } from "react";
import styles from "./SearchTask.module.scss";

type SearchTaskProps = {
  searchText: string;
  setSearchText: Dispatch<SetStateAction<string>>;
  setIsSortByCreatedAt: Dispatch<SetStateAction<boolean>>;
};

export default function SearchTask({
  searchText,
  setSearchText,
  setIsSortByCreatedAt,
}: SearchTaskProps) {
  return (
    <div className={styles["search-task-wrap"]}>
      <input
        type="text"
        value={searchText}
        onChange={(e) => setSearchText(e.target.value)}
        placeholder="Search tasks"
        className={styles["search-task-input"]}
      />
      <div className={styles["checkbox-wrap"]}>
        <input
          id="sort-by-created-date"
          type="checkbox"
          onChange={(e) => setIsSortByCreatedAt(e.target.checked)}
          name="sort-by-created-date"
          className={styles["checkbox-input"]}
        />
        <label
          htmlFor="sort-by-created-date"
          className={styles["checkbox-label"]}
        >
          Sort by created date
        </label>
      </div>
    </div>
  );
}
