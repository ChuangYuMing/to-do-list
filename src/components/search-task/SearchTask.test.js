import React from "react";
import { render } from "@testing-library/react";
import SearchTask from "./SearchTask";

it("renders correctly", () => {
  const { asFragment } = render(<SearchTask />);
  expect(asFragment()).toMatchSnapshot();
});
