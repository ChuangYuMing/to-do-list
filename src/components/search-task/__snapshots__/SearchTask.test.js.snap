// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-task-wrap"
  >
    <input
      class="search-task-input"
      placeholder="Search tasks"
      type="text"
      value=""
    />
    <div
      class="checkbox-wrap"
    >
      <input
        class="checkbox-input"
        id="sort-by-created-date"
        name="sort-by-created-date"
        type="checkbox"
      />
      <label
        class="checkbox-label"
        for="sort-by-created-date"
      >
        Sort by created date
      </label>
    </div>
  </div>
</DocumentFragment>
`;
